sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/SpanDataAssert.java

```diff
@@ -112,6 +112,11 @@ public SpanDataAssert hasParentSpanId(String parentSpanId) {
     return this;
   }
 
+  /** Asserts the span has the given parent {@link SpanData span}. */
```

Comment:
I believe that this method is slightly misleading, as it checks only if current span has parentId equal to argument's id. It does not check whole span (all attributes). It does yield shorter, more readable code so perhaps should have a different name to avoid any confusion (say `hasParentId` - overloaded method) ?

Reply:
What does it mean to check the whole span in this case? I feel it's pretty accurate even if just checking id

Reply:
If I were an average Joe user I'd assume that hasParentId() will check if parent of the current span has the specified ID, while hasParent(span) will check that the parent is a particular span (all attributes matching). So in order to avoid confusion, I believe that this information (only ID is asserted) could be included either in name or in javadoc of this method.

Reply:
I think the testing artifact isn't for average Joes so the APIs don't have to be as simple, there's a stronger emphasis on convenience to make sure people write instrumentation I think. But more docs is always good, I added the code snippet of equivalence does it help clear it up?

Reply:
Definitely! Thanks :)